### ISSUE TITLE
Whisper generation support for passing acronym to language arg

### ISSUE THREAD

**Issue by connor-henderson** [huggingface/transformers#23140]
### System Info

- `transformers` version: 4.29.0.dev0
- Platform: macOS-13.0-arm64-arm-64bit
- Python version: 3.9.16
- Huggingface_hub version: 0.12.0
- Safetensors version: 0.2.8
- PyTorch version (GPU?): 1.13.1 (False)
- Tensorflow version (GPU?): 2.11.0 (False)
- Flax version (CPU?/GPU?/TPU?): 0.5.3 (cpu)
- Jax version: 0.3.6
- JaxLib version: 0.3.5
- Using GPU in script?: No
- Using distributed or parallel set-up in script?: No

### Who can help?

@hollance @gante

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [X] My own task or dataset (give details below)

### Reproduction

```py
processor = WhisperProcessor.from_pretrained("openai/whisper-tiny")
model = WhisperForConditionalGeneration.from_pretrained("openai/whisper-tiny")
ds = load_dataset("hf-internal-testing/librispeech_asr_dummy", "clean", split="validation")
sample = ds[0]["audio"]["array"]
input_features = processor.feature_extractor(sample, return_tensors="pt").input_features

pred_ids = model.generate(input_features, language="de")
```

### Expected behavior

Should recognize and use language passed in acronym format as per the docstring

**Comment by gante** [comment]
cc @ArthurZucker

**Comment by hollance** [comment]
Yeah I'm not sure why it was decided the language token had to be passed in there, and at the very least the current error message is misleading. Arthur is probably the best person to look at this.
